opensrp-reveal/src/main/java/org/smartregister/reveal/fragment/BaseOfflineMapsFragment.java

```diff
@@ -0,0 +1,142 @@
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v4.content.LocalBroadcastManager;
+import android.text.TextUtils;
+import android.util.Log;
+import android.widget.Toast;
+
+import org.smartregister.reveal.R;
+import org.smartregister.reveal.contract.OfflineMapsFragmentContract;
+
+import io.ona.kujaku.services.MapboxOfflineDownloaderService;
+import io.ona.kujaku.utils.Constants;
+
+public class BaseOfflineMapsFragment extends Fragment implements OfflineMapsFragmentContract.View {
+
+
+    private MapDownloadReceiver mapDownloadReceiver = new MapDownloadReceiver();
+
+    protected String currentMapDownload;
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        IntentFilter intentFilter = new IntentFilter(Constants.INTENT_ACTION_MAP_DOWNLOAD_SERVICE_STATUS_UPDATES);
+        LocalBroadcastManager.getInstance(getActivity().getApplicationContext()).registerReceiver(mapDownloadReceiver, intentFilter);
+    }
+
+    @Override
+    public void onPause() {
+        super.onPause();
+        LocalBroadcastManager.getInstance(getActivity().getApplicationContext()).unregisterReceiver(mapDownloadReceiver);
+    }
+
+    @Override
+    public void displayToast(String message) {
+        Toast.makeText(getContext(), message, Toast.LENGTH_LONG).show();
+    }
+
+    @Override
+    public void displayError(int title, String message) {
+        Toast.makeText(getContext(), message, Toast.LENGTH_LONG).show();
+    }
+
+    private class MapDownloadReceiver extends BroadcastReceiver {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            Bundle bundle = intent.getExtras();
+            if (bundle != null) {
+                Log.i("KUJAKU SAMPLE APP TAG", intent.getExtras().toString());
+                if (bundle.containsKey(MapboxOfflineDownloaderService.KEY_RESULT_STATUS)
+                        && bundle.containsKey(MapboxOfflineDownloaderService.KEY_RESULT_MESSAGE)
+                        && bundle.containsKey(MapboxOfflineDownloaderService.KEY_RESULTS_PARENT_ACTION)
+                        && bundle.containsKey(Constants.PARCELABLE_KEY_MAP_UNIQUE_NAME)) {
+
+                    String mapUniqueName = bundle.getString(Constants.PARCELABLE_KEY_MAP_UNIQUE_NAME);
+                    String resultStatus = bundle.getString(MapboxOfflineDownloaderService.KEY_RESULT_STATUS);
+                    MapboxOfflineDownloaderService.SERVICE_ACTION serviceAction = (MapboxOfflineDownloaderService.SERVICE_ACTION) bundle.get(MapboxOfflineDownloaderService.KEY_RESULTS_PARENT_ACTION);
+
+                    String message = bundle.getString(MapboxOfflineDownloaderService.KEY_RESULT_MESSAGE);
+
+                    if (MapboxOfflineDownloaderService.SERVICE_ACTION_RESULT.FAILED.name().equals(resultStatus)) {
+                        if (!TextUtils.isEmpty(message)) {
+                            if (!message.contains("MapBox Tile Count limit exceeded")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=4217760)